data/moves.js

```diff
@@ -14757,10 +14757,11 @@ let BattleMovedex = {
 		beforeTurnCallback(pokemon) {
 			pokemon.addVolatile('shelltrap');
 		},
-		beforeMoveCallback(pokemon) {
+		onTry(pokemon) {
 			if (pokemon.volatiles['shelltrap'] && !pokemon.volatiles['shelltrap'].gotHit) {
```

Comment:
```suggestion
			if (!pokemon.volatiles['shelltrap'] || !pokemon.volatiles['shelltrap'].gotHit) {
```
This should make Shell Trap fail correctly when forced to be used from a decision-changing Encore.

Reply:
This didn't seem to fix things? But maybe my test is wrong.

Reply:
My test was wrong. Thanks for the fix!